tikv-client/src/main/java/com/pingcap/tikv/expression/ComparisonBinaryExpression.java

```diff
@@ -0,0 +1,199 @@
+    return new ComparisonBinaryExpression(LESS_EQUAL, left, right);
+  }
+
+  public static ComparisonBinaryExpression greaterThan(Expression left, Expression right) {
+    return new ComparisonBinaryExpression(GREATER_THAN, left, right);
+  }
+
+  public static ComparisonBinaryExpression greaterEqual(Expression left, Expression right) {
+    return new ComparisonBinaryExpression(GREATER_EQUAL, left, right);
+  }
+
+  public static class NormalizedPredicate {
+    private final ComparisonBinaryExpression pred;
+    private TypedKey key;
+
+    NormalizedPredicate(ComparisonBinaryExpression pred) {
+      checkArgument(pred.getLeft() instanceof ColumnRef);
+      checkArgument(pred.getRight() instanceof Constant);
+      this.pred = pred;
+    }
+
+    public ColumnRef getColumnRef() {
+      return (ColumnRef) pred.getLeft();
+    }
+
+    public Constant getValue() {
+      return (Constant) pred.getRight();
+    }
+
+    public Type getType() {
+      return pred.getComparisonType();
+    }
+
+    public TypedKey getTypedLiteral() {
+      if (key == null) {
+        key = TypedKey.toTypedKey(getValue().getValue(), getColumnRef().getType());
+      }
+      return key;
+    }
+  }
+
+  private final Expression left;
+  private final Expression right;
+  private final Type compType;
+  transient private Optional<NormalizedPredicate> normalizedPredicate;
+
+  public ComparisonBinaryExpression(Type type, Expression left, Expression right) {
+    this.left = requireNonNull(left, "left expression is null");
+    this.right = requireNonNull(right, "right expression is null");
+    this.compType = requireNonNull(type, "type is null");
+  }
+
+  @Override
+  public List<Expression> getChildren() {
+    return ImmutableList.of(left, right);
+  }
+
+  @Override
+  public <R, C> R accept(Visitor<R, C> visitor, C context) {
+    return visitor.visit(this, context);
+  }
+
+  public Expression getLeft() {
+    return left;
+  }
+
+  public Expression getRight() {
+    return right;
+  }
+
+  public Type getComparisonType() {
+    return compType;
+  }
+
+  public NormalizedPredicate normalize() {
+    if (normalizedPredicate != null) {
+      return normalizedPredicate.orElseGet(() -> null);
+    }
+    if (getLeft() instanceof Constant && getRight() instanceof ColumnRef) {
+      Constant left = (Constant) getLeft();
+      ColumnRef right = (ColumnRef) getRight();
+      Type newType;
+      switch (getComparisonType()) {
+        case EQUAL:
+          newType = EQUAL;
+          break;
+        case LESS_EQUAL:
+          newType = Type.GREATER_EQUAL;
+          break;
+        case LESS_THAN:
+          newType = Type.GREATER_THAN;
+          break;
+        case GREATER_EQUAL:
+          newType = Type.LESS_EQUAL;
+          break;
+        case GREATER_THAN:
+          newType = Type.LESS_THAN;
+          break;
+        case NOT_EQUAL:
+          newType = NOT_EQUAL;
```

Comment:
`Type.`, maybe stay consistent with other calls' is better.

Reply:
Unified.